### ISSUE TITLE
No parameter named 'provider'

### ISSUE THREAD

**Issue by ngeorgieff** [antoineco/aco-oracle_java#19]
I'm getting "no parameter named 'provider' if I run puppet agent -t but it works if I use puppet apply.
PE 2016.2
RHEL 7

```
[root@is-engineering-t01 manifests]# puppet agent -t
Notice: Local environment: 'production' doesn't match server specified node environment 'lab', switching agent to 'lab'.
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: no parameter named 'provider' at /etc/puppetlabs/code/environments/lab/modules/oracle_java/manifests/download.pp:42 on Archive[/usr/java/jre-8u92-linux-x64.tar.gz] at /etc/puppetlabs/code/environments/lab/modules/oracle_java/manifests/download.pp:42 on node is-engineering-t01.it.****.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
```
```
[root@is-engineering-t01 manifests]# puppet facts | grep aio
    "aio_agent_build": "1.4.2",
    "aio_agent_version": "1.4.2",
```

**Comment by antoineco** [comment]
I think the fact that you're using Puppet **Enterprise** makes a real difference. Gems are not installed in the same manner in Puppet OSS and Enterprise.

Check this out: https://github.com/voxpupuli/puppet-archive/pull/121
And the deletions in the `params.pp` file here: https://github.com/voxpupuli/puppet-archive/commit/424920045916af95d0a20b5ed5022b721a2f3876#diff-2f383a197b1a428561b364755701d60e

It works just fine with Puppet OSS:

```
root@62a78ffc135f:/# export PATH=/opt/puppetlabs/puppet/bin:$PATH

root@62a78ffc135f:/# puppet module install aco-oracle_java

Notice: Preparing to install into /etc/puppetlabs/code/environments/production/modules ...
Notice: Downloading from https://forgeapi.puppetlabs.com ...
Notice: Installing -- do not interrupt ...
/etc/puppetlabs/code/environments/production/modules
└─┬ aco-oracle_java (v2.6.3)
  ├─┬ puppet-archive (v0.4.8)
  │ └── puppetlabs-pe_gem (v0.2.0)
  └── puppetlabs-stdlib (v4.12.0)

root@62a78ffc135f:/# puppet apply -t -e 'include oracle_java'

Info: Loading facts
Info: Loading facts
Notice: Compiled catalog for 62a78ffc135f.lan in environment production in 0.54 seconds
Info: Applying configuration version '1467061100'
Notice: /Stage[main]/Archive/Package[faraday]/ensure: created
Notice: /Stage[main]/Archive/Package[faraday_middleware]/ensure: created
Notice: /Stage[main]/Oracle_java::Download/File[/usr/java]/ensure: created
Notice: /Stage[main]/Oracle_java::Download/Archive[/usr/java/jre-8u92-linux-x64.tar.gz]/ensure: download archive from http://download.oracle.com/otn-pub/java/jdk/8u92-b14/jre-8u92-linux-x64.tar.gz to /usr/java/jre-8u92-linux-x64.tar.gz and extracted in /usr/java with cleanup
Info: Class[Oracle_java::Download]: Scheduling refresh of Class[Oracle_java::Install]
Info: Class[Oracle_java::Install]: Scheduling refresh of Class[Oracle_java::Install::Targz]
Notice: /Stage[main]/Oracle_java::Install::Targz/File[/usr/java/default]/ensure: created
Info: Creating state file /opt/puppetlabs/puppet/cache/state/state.yaml
Notice: Applied catalog in 66.73 seconds
```

**Comment by antoineco** [comment]
@ngeorgieff this should be fixed in `puppet/archive` v1.0.0. Please also update to `aco/oracle_java` v2.7.0 if possible, to avoid warnings on dependency versions.

Could you please confirm whenever you find the time to test it?

**Comment by ngeorgieff** [comment]
@antoineco Thank you very much for your help. I will update to the latest version.
